internal/saucecloud/cloud.go

```diff
@@ -146,30 +143,14 @@ func (r *CloudRunner) collectResults(artifactCfg config.ArtifactDownload, result
 			}
 
 			var artifacts []report.Artifact
-			var parentJUnits []report.Artifact
-
-			if junitRequired {
-				jb, err := r.JobService.GetJobAssetFileContent(
-					context.Background(),
-					res.job.ID,
-					junit.JunitFileName,
-					res.job.IsRDC)
-				artifacts = append(artifacts, report.Artifact{
-					AssetType: report.JUnitArtifact,
-					Body:      jb,
-					Error:     err,
-				})
-				for _, id := range res.previous {
-					jb, err := r.JobService.GetJobAssetFileContent(
-						context.Background(),
-						id,
-						junit.JunitFileName,
-						res.job.IsRDC,
-					)
-					parentJUnits = append(parentJUnits, report.Artifact{
-						AssetType: report.JUnitArtifact,
-						Body:      jb,
-						Error:     err,
+			junitArtifact, parentJUnits := r.GetJUnitArtifacts(res)
+			artifacts = append(artifacts, junitArtifact)
+
+			files := r.downloadArtifacts(res.name, res.job, artifactCfg.When)
+			if report.IsArtifactRequired(r.Reporters, report.JSONArtifact) {
+				for _, f := range files {
+					artifacts = append(artifacts, report.Artifact{
+						FilePath: f,
 					})
 				}
 			}
```

Comment:
Maybe this was already a code smell before, but shouldn't we just do this (sorry, my formatting might be messed up in the suggestion box)?

```suggestion
			files := r.downloadArtifacts(res.name, res.job, artifactCfg.When)
		              for _, f := range files {
			              artifacts = append(artifacts, report.Artifact{
				              FilePath: f,
			              })
		              }
			}
```